ethereum/contracts/relayer/coreRelayer/Utils.sol

```diff
@@ -39,3 +39,53 @@ function fromWormholeFormat(bytes32 whFormatAddress) pure returns (address) {
+uint256 constant maskModulo32 = 0x1f;
+// Bound chosen by the following formula: `memoryWord * 4 + selectorSize`.
+// This means that an error identifier plus four fixed size arguments should be available to developers.
+// In the case of a `require` revert with error message, this should provide 3 memory word's worth of data.
+uint256 constant returnLengthBound = 132;
+
+/**
+ * Implements call that truncates return data to a constant size to avoid excessive gas consumption for relayers
+ * when a revert or .
+ */
+function returnLengthBoundedCall(address payable callee, bytes memory callData, uint256 gasLimit, uint256 value) returns (bool success, bytes memory returnedData) {
+  uint256 callDataLength = callData.length;
+  assembly ("memory-safe") {
+    returnedData := mload(freeMemoryPtr)
+    // Note that `returnedDataEndIndex` and `callDataEndIndex` are past the end indexes for their respective buffers.
+    let returnedDataBuffer := add(returnedData, memoryWord)
+    let returnedDataEndIndex := add(returnedDataBuffer, returnLengthBound)
+    let callDataStartIndex := add(callData, memoryWord)
+    let callDataEndIndex := add(callDataStartIndex, callDataLength)
+
+    success := call(gasLimit, callee, value, callDataStartIndex, callDataEndIndex, returnedDataBuffer, returnedDataEndIndex)
+    let returnedDataSize := returndatasize()
+    switch lt(returnedDataSize, add(returnLengthBound, 1))
+    case 0 {
+      returnedDataSize := returnLengthBound
+    } default {}
+    mstore(returnedData, returnedDataSize)
+
+    // Here we update the free memory pointer.
+    // We want to pad `returnedData` to memory word size, i.e. 32 bytes.
+    // Note that negating bitwise `maskModulo32` produces a mask that aligns addressing to 32 bytes.
+    // This allows us to pad the entire `bytes` structure (length + buffer) to 32 bytes at the end.
+    // We add `maskModulo32` to get the next free memory "slot" in case the `returnedDataSize` is not a multiple of the memory word size.
+    //
+    // Rationale:
+    // We do not care about the alignment of the free memory pointer. The solidity compiler documentation does not promise nor require alignment on it.
+    // It does however lightly suggest to pad `bytes` structures to 32 bytes: https://docs.soliditylang.org/en/v0.8.20/assembly.html#example
+    // Searching for "alignment" and "padding" in https://gitter.im/ethereum/solidity-dev
+    // yielded the following at the time of writing – paraphrased:
+    // > It's possible that the compiler cleans that padding in some cases. Users should not rely on the compiler never doing that.
+    // This means that we want to ensure that the free memory pointer points to memory just after this padding for our `returnedData` `bytes` structure.
+    let paddedPastTheEndOffset := and(add(returnedDataSize, maskModulo32), not(maskModulo32))
+    let newFreeMemoryPtr := add(returnedDataBuffer, paddedPastTheEndOffset)
+    mstore(freeMemoryPtr, newFreeMemoryPtr)
```

Comment:
love the doc and the efficiency (compared to the lazy implementation of using `new bytes(lengthBound)` before and a `.slice(actualLength)` after)